docs: added docs on localization & culture hotswapping (#5512)
unoplatform/uno: doc/articles/guides/localization-and-culture-hot-swapping.md

# How to use localization and perform runtime culture switching
## Full Sample

> Both complete sample code can be found here:

Reviewer: Modify the intro to match the template: https://github.com/unoplatform/uno/blame/dev/djo/docs-fix-3/doc/.howto-template.md

- [Localization](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/Localization)
- [RuntimeCultureSwitching](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/RuntimeCultureSwitching)

## Localization

1. Create a new Uno Platform application, following the instructions [here](..\get-started.md).

> 1. Modify the content of `MainPage` a bit:

Reviewer: ```suggestion
1. Modify the content of `MainPage`:
```

    - In `MainPage.xaml`, replace the content of the page:
        ```xml
        <Page x:Class="UnoLocalization.MainPage"
              xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
              xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
              xmlns:toolkit="using:Uno.UI.Toolkit">

            <StackPanel toolkit:VisibleBoundsPadding.PaddingMask="Top">
                <TextBlock x:Uid="MainPage_IntroText" Text="Hello, world!" Margin="20" FontSize="30" />
                <TextBlock x:Name="CodeBehindText" Text="This text will be replaced" />
            </StackPanel>
        </Page>
        ```

>         > note:

Reviewer: ```suggestion
        > [!NOTE]
```

        > - The [`x:Name`](https://docs.microsoft.com/en-us/windows/uwp/xaml-platform/x-name-attribute) is used to make the element accessible from the code-behind with that same name.
        > - The [`x:Uid`](https://docs.microsoft.com/en-us/windows/uwp/xaml-platform/x-uid-directive) is used for localization.
        To localize a property, you need to add a string resource in each resource file using its `x:Uid` followed by a dot (`.`) and then the property name. eg: `MainPage_IntroText.Text`
        More on this in the resource steps that follow.
    - In `MainPage.xaml.cs`, add an `Page.Loaded` handler to change the text for `CodeBehindText`:
        ```cs
        public MainPage()
        {
            this.InitializeComponent();
            this.Loaded += (s, e) => CodeBehindText.Text = Windows.ApplicationModel.Resources.ResourceLoader
                .GetForViewIndependentUse()
                .GetString("MainPage_CodeBehindString");
        }
        ```

> 1. Create a new resource file for localization in french in the `UnoLocalization.Shared` project:

Reviewer: ```suggestion
1. Create a new resource file for localization in French in the `UnoLocalization.Shared` project:
```
```suggestion
1. Create a new resource file for localization in french in the `UnoLocalization.Shared` project:
```

    1. Add a new folder `fr` under the `Strings` folder by:
    R-Click on `String` > Add > New Folder
    1. Add a new resource file `Resources.resw` under the `fr` folder by:
    R-Click on `fr` > Add > New Item ... > Visual C# > Xaml > Resources File

> 1. Add some localization texts for both english and french:

Reviewer: ```suggestion
1. Add some localized text entries for both English and French:
```

    Open both `Strings\en\Resources.resw` and `Strings\fr\Resources.xaml`, and add these:
    Name|Value in en\Resources.resw|Value in fr\Resources.resw
    -|-|-
    MainPage_IntroText.Text|Hello Uno|Bonjour Uno
    MainPage_CodeBehindString|String from code-behind|Texte provenant du code-behind
    > note: Be sure to hit Ctrl+S for both files, to save the changes.
1. You can now try to run the app.

    The "Hello World" text should be replaced with "Hello Uno", or "Bonjour Uno" if the targeted environment is on french culture.
    Now, if you change the language of the targeted PC or the mobile device AND restart the app, that text should also change accordingly.

    You can also set the starting culture without affecting the OS settings using this method:
    - `App.xaml.cs`:
        ```cs
        public App()
        {
            ConfigureFilters(global::Uno.Extensions.LogExtensionPoint.AmbientLoggerFactory);
            ChangeStartingLanguage();

            this.InitializeComponent();
            this.Suspending += OnSuspending;
        }

        private void ChangeStartingLanguage()
        {
            var culture = new System.Globalization.CultureInfo("fr");

>         #if WINDOWS_UWP

Reviewer: per the discussion on the other PR, I think you can safely use this Uno 3.6 feature

            Windows.Globalization.ApplicationLanguages.PrimaryLanguageOverride = culture.IetfLanguageTag;
        #else
            System.Globalization.CultureInfo.CurrentUICulture = culture;
            System.Globalization.CultureInfo.CurrentCulture = culture;
            Windows.ApplicationModel.Resources.ResourceLoader.DefaultLanguage = culture.IetfLanguageTag;
        #endif
        }
        ```

> ## Runtime culture switching

Reviewer: Put this in a separate article

Author: done
